src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -2078,7 +2078,7 @@ internal static BinaryOperationKind DeriveBinaryOperationKind(UnaryOperationKind
                     return BinaryOperationKind.DynamicSubtract;
 
                 default:
-                    throw ExceptionUtilities.UnexpectedValue(incrementKind);
+                    return BinaryOperationKind.None;
```

Comment:
I found an example of error value as the last entry in enum though. 
```
Microsoft.CodeAnalysis.VisualBasic.UnaryOperatorKind.Error
```

And FxCop suggests that the none value should indicate that no values have been set in the enum, which I think is not the case for Invalid. But yea, I can leave it as None today :)